src/udev/udev-builtin-net_id.c

```diff
@@ -341,18 +343,24 @@ static int dev_pci_slot(sd_device *dev, struct netnames *names) {
 
         /* kernel provided port index for multiple ports on a single PCI function */
         if (sd_device_get_sysattr_value(dev, "dev_port", &attr) >= 0) {
-                dev_port = strtoul(attr, NULL, 10);
+                r = safe_atolu_full(attr, 10, &dev_port);
+                if (r < 0)
+                        log_device_debug_errno(dev, r, "Failed to parse dev_port, ignoring: %m");
                 /* With older kernels IP-over-InfiniBand network interfaces sometimes erroneously
                  * provide the port number in the 'dev_id' sysfs attribute instead of 'dev_port',
                  * which thus stays initialized as 0. */
                 if (dev_port == 0 &&
                     sd_device_get_sysattr_value(dev, "type", &attr) >= 0) {
                         unsigned long type;
 
-                        type = strtoul(attr, NULL, 10);
+                        r = safe_atolu_full(attr, 10, &type);
+                        if (r < 0)
+                                log_device_debug_errno(dev, r, "Failed to parse type, ignoring: %m");
                         if (type == ARPHRD_INFINIBAND &&
-                            sd_device_get_sysattr_value(dev, "dev_id", &attr) >= 0)
-                                dev_port = strtoul(attr, NULL, 16);
+                            sd_device_get_sysattr_value(dev, "dev_id", &attr) >= 0) {
+                                r = safe_atolu_full(attr, 10, &dev_port);
+                                log_device_debug_errno(dev, r, "Failed to parse dev_id, ignoring: %m");
```

Comment:
missing `if (r < 0)` ? Also it's dev_port being parsed

Reply:
> missing `if (r < 0)` ? Also it's dev_port being parsed

Yes, but the attribute that's parsed is called `dev_id` in this case (as opposed to what's parsed a couple of lines above). Maybe I should put the word "attribute" into the message to make that clear?

Reply:
Right. `dev_id` in log message is correct.

> Maybe I should put the word "attribute" into the message to make that clear?

Yeah. That sounds nicer.